src/include/sof/audio/format.h

```diff
@@ -79,6 +79,13 @@
 #define SATP_INT32(x) (((x) > INT32_MAX) ? INT32_MAX : (x))
 #define SATM_INT32(x) (((x) < INT32_MIN) ? INT32_MIN : (x))
 
+/* Arithmetic shift right */
+#define ARITHMETIC_SHIFT_RIGHT(u, n) ({	\
+	(u > 0) ?	\
+		(int32_t)((uint32_t)(((uint32_t)u) >> ((uint32_t)n))) :	\
+		(-((int32_t)((uint32_t)(((uint32_t)((int32_t)(-1 - u))) >> ((uint32_t)n))))) - 1; \
+})
```

Comment:
Please see my previous comments. This needs to be a static inline function which is type safe.